### ISSUE TITLE
Should actions be invoked and checked as this.attrs.action

### ISSUE THREAD

**Issue by cah-briangantzler** [cibernox/ember-power-select#233]
Just noticed in the for initialization that the check is done at didInitAttrs, but they could have passed in an action from a variable that could be changed causing the component to be re-rendered. Should didReceiveAttrs be used instead?

Also this.attrs.onchange should be used and no depend on the current mirroring to root space. Hmmm, i'm looking at code, should all the invocations of actions be this.attrs.action() instead of this.get("action")().

some thoughts before you put another beta out

should this
```
didInitAttrs() {
    this._super(...arguments);
    Ember.assert('{{power-select}} requires an `onchange` function', this.get('onchange') && typeof this.get('onchange') === 'function');
  },
```
be this
```
didReceiveAttrs() {
    this._super(...arguments);
    Ember.assert('{{power-select}} requires an `onchange` function', this.attrs.onchange && typeof this.attrs.onchange === 'function');
  },
```

**Comment by cibernox** [comment]
About `this.attrs.foobar()` vs `this.get('foobar')()`, I've been advised by "the wise elder" in the core team to wait to glimmer components before using `this.attrs`. The argument they gave me is that at the moment they are a bit inconsistent because with double bindings you need to read the values with `this.attrs.foobar.value` (because this.attrs.foobar is a mutable stream) while if the user uses `foobar=(readonly foo)` or `foobar="someStaticValue"` you will access it `this.attrs.foobar` because is immutable, and at the moment closure actions are treated like an exception to this rule.

Glimmer components will treat everything as readonly, so this inconsistency will dissapear.

About the warning, I used `didInitAttr` because `didReceiveAttrs` because this was just a warning for development, mostly for help newbies to undestand how the component works (many people still expects double bindings).
I just never considered that someone could swap the action in runtime to something that is null. Do you think it works changing it'?

**Comment by cah-briangantzler** [comment]
there is a helper method `this.getAttr()` that handles the direct or .value problem. But if you were advised by core, I would go with that then.

Thanks

**Comment by cibernox** [comment]
I wasn't aware of that `getAttr` method, and I can't find it on the API docs. 

If it's available in 1.13 I might prefer that over get, because even if it is not a direct property access, it keeps the distintion between own vs passed-in properties, which is something I'd like to have and with `get` you loose that concept.

**Comment by cah-briangantzler** [comment]
Yes its in 1.13
http://discuss.emberjs.com/t/boolean-component-attributes-in-attrs-hash/8029
